### PR TITLE
Fix #1520: Misleading type names

### LINKED ISSUE #1520: Misleading type names
The current names of the basic underlying types are misleading. These should be changed to make the code easier to understand. The changes should be made as follows:

- `pyccel.ast.basic.Basic` -> `pyccel.ast.basic.PyccelAstNode`
- `pyccel.ast.basic.PyccelAstNode` -> `pyccel.ast.basic.TypedAstNode`
- `pyccel.ast.basic.ScopedNode` -> `pyccel.ast.basic.ScopedAstNode`

### PR DESCRIPTION
**Pull request by imperial-chief**
First replaced ScopedNode to ScopedAstNode
Then PyccelAstNode to TypedAstNode
and lastly Basic to PyccelAstNode.

### CODE REVIEW

File: container/bin/python3.11 [hunk 1]
```diff
@@ -0,0 +1 @@
+python3
```
Comment (EmilyBourne) [inline container/bin/python3.11:1]: This folder seems to have been added by accident

Comment (imperial-chief) [reply container/bin/python3.11:1]: yeah the container folder was added, when i tested

File: pyccel/ast/core.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def subs(expr, new_elements):
         new_expr = expr.subs(new_elements)
         new_expr.set_fst(expr.fst)
         return new_expr
-    elif isinstance(expr, PyccelAstNode):
+    elif isinstance(expr, TypedAstNode):
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/core.py:164]: /bot accept
This is known dead code that we need to remove

**Comment by EmilyBourne** [comment]
@yguclu Can we make an exception for these documentation errors. They are related to Numpydoc problems but there are **a lot**. I think this change is important and is better done sooner rather than later so I think it is worth making an exception here

**Review by EmilyBourne — APPROVED** [review]
This looks good. Well done on finding them all and thanks for taking care of this for us :)

File: pyccel/ast/basic.py [hunk 3]
```diff
@@ -6,14 +6,14 @@
 """
 This module contains classes from which all pyccel nodes inherit.
 They are:
-- Basic, which provides a python AST
-- PyccelAstNode which describes each PyccelAstNode
+- PyccelAstNode, which provides a python AST
+- TypedAstNode which describes each TypedAstNode
```
Comment (yguclu) [inline pyccel/ast/basic.py:10]: Not sure what this line was supposed to read...

Comment (EmilyBourne) [reply pyccel/ast/basic.py:10]: Hmmm, this is definitely unreadable. I suggest:
``` 
- PyccelAstNode, which provides a base class for our Python AST nodes
 - TypedAstNode which provides a base class for AST nodes which require type descriptors 
```

Comment (yguclu) [reply pyccel/ast/basic.py:10]: Almost perfect. Isn't `TypedAstNode` a subclass of `PyccelAstNode`?

Comment (EmilyBourne) [reply pyccel/ast/basic.py:10]: Yes
```
- PyccelAstNode, which provides a base class for our Python AST nodes
- TypedAstNode which inherits from PyccelAstNode and provides a base class for AST nodes which require type descriptors 
```

Comment (yguclu) [reply pyccel/ast/basic.py:10]: Please apply the change!

Comment (yguclu) [reply pyccel/ast/basic.py]: Done in 1c96320.

**Pull request merged**